tests/integ/sagemaker/jumpstart/retrieve_uri/utils.py

```diff
@@ -0,0 +1,116 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import uuid
+from typing import Match, Optional, Tuple
+import boto3
+import pandas as pd
+import regex
+import os
+
+from tests.integ.sagemaker.jumpstart.retrieve_uri.constants import (
+    HYPERPARAMETER_MODEL_DICT,
+    TEST_ASSETS_SPECS,
+    TMP_DIRECTORY_PATH,
+    TRAINING_DATASET_MODEL_DICT,
+)
+from sagemaker.jumpstart.constants import JUMPSTART_DEFAULT_REGION_NAME
+from sagemaker.jumpstart.utils import get_jumpstart_content_bucket
+
+from sagemaker.s3 import parse_s3_url
+from sagemaker.session import Session
+
+
+def download_file(local_download_path, s3_bucket, s3_key, s3_client) -> None:
+    s3_client.download_file(s3_bucket, s3_key, local_download_path)
+
+
+def extract_role_arn_from_caller_identity(caller_identity_arn: str) -> str:
+
+    ASSUME_ROLE_REGEX = r"^(?P<prefix>.+)sts::(?P<infix>\d+):assumed-role\/(?P<suffix>.+?)\/.*$"
+
+    match: Optional[Match[str]] = regex.match(ASSUME_ROLE_REGEX, caller_identity_arn)
+
+    if match is None:
+        # not an assumed role caller identity
+        return caller_identity_arn
+
+    prefix, infix, suffix = match.groups()
+
+    return f"{prefix}iam::{infix}:role/{suffix}"
+
+
+def get_model_tarball_full_uri_from_base_uri(base_uri: str, training_job_name: str) -> str:
+    return os.path.join(
+        base_uri,
+        training_job_name,
+        "output",
+        "model.tar.gz",
+    )
+
+
+def get_full_hyperparameters(
+    base_hyperparameters: dict, job_name: str, model_artifacts_uri: str
+) -> dict:
+
+    bucket, key = parse_s3_url(model_artifacts_uri)
+    return {
+        **base_hyperparameters,
+        "sagemaker_job_name": job_name,
+        "model-artifact-bucket": bucket,
+        "model-artifact-key": key,
+    }
+
+
+def get_hyperparameters_for_model_and_version(model_id: str, version: str) -> dict:
+    return HYPERPARAMETER_MODEL_DICT[(model_id, version)]
+
+
+def get_training_dataset_for_model_and_version(model_id: str, version: str) -> dict:
+    return TRAINING_DATASET_MODEL_DICT[(model_id, version)]
+
+
+def get_test_cache_bucket() -> str:
```

Comment:
non-blocking: "test cache bucket" is a bit of a misnomer, it is the bucket where the artifacts supporting the SageMaker resources you deploy are saved.